### ISSUE TITLE
Question about inference speed

### ISSUE THREAD

**Issue by Pixie8888** [YuDeng/Portrait-4D#12]
Dear authors,

You mentioned inference speed is 10 fps in the paper. Do you include the time of ```full BFM-to-FLAME transformation process```? In my machine, it takes more than 1 minute to generate a result (include the time for Landmark detection, 3D face reconstruction, and cropping, BFM to FLAME parameter transformation).

I also have a question about training portrait4d-v2. Did the 3D synthesiser generate multi-view driving images on the fly during training shown in the fig 2? Could you please help me point out where is the relevant code?

**Comment by YuDeng** [comment]
Hi, the inference time does not take FLAME optimization into consideration. 

For training potrait4d-v2, multi-view images are generating online via the 3D synthesizer. Relevant codes are here: https://github.com/YuDeng/Portrait-4D/blob/da5eec6fa3dfca4d7c8f08daa46326f2de8244db/portrait4d/training/loss/loss_recon_v2.py#L233; https://github.com/YuDeng/Portrait-4D/blob/da5eec6fa3dfca4d7c8f08daa46326f2de8244db/portrait4d/training/loss/loss_recon_v2.py#L336.

**Comment by Pixie8888** [comment]
Thank you for your reply! I want to try training Portrait4d v2 on the toy dataset on a 24 GB gpu. I set the ```batch = 1``` in ```portrait4d-v2-vfhq512-toy.yaml```, but it shows error below:
![image](https://github.com/user-attachments/assets/44ef9306-ae0f-438c-86a8-d3b6cc92010b)


How can I train the model with batchsize of 1?

---------------------------- update -------------------------------
I changed ```mbstd_group = 1``` in ```default.yaml``` and ```batch=1``` in portrait4d_v2_vfhq512_toy.yaml``` to train on 24 GB gpu. My question is changing batchsize to 1 will affect model's performance?  Particularly what is the use of ```mbstd_group```?

**Comment by YuDeng** [comment]
mbstd_group is a standard operation inherited from StyleGAN's discriminator to calculate some statistic values of real and fake data. A small mbstd_group may have slight influence to the performance of GAN loss.

**Comment by Pixie8888** [comment]
Thank you for your reply. I have some other question regarding evaluation. Could you please share the code on how to compute the evaluation metric in Table 1? And the video ids that used for evaluation?

**Comment by YuDeng** [comment]
Sorry that the evaluation part is not available. I've switched job recently and is unable to reach the original code.

**Comment by longyangqi** [comment]
> Hi, the inference time does not take FLAME optimization into consideration.

I want to know the usage of the FLAME params in the reenactment inference part, while the FLAME optimization takes a long time.
In fig2 of the paper Portrait4Dv2, I cannot find where the FLAME is used in. And from the code, I find it is used to construct the Deformer, but I do not know the deformer is built for what.
Moreover, if we only want to get certain view(e.g. just the front view) from reference image driven by target image, do i still need the FLAME params? If not, the process would be much faster.
Thanks!

**Comment by YuDeng** [comment]
> > Hi, the inference time does not take FLAME optimization into consideration.
> 
> I want to know the usage of the FLAME params in the reenactment inference part, while the FLAME optimization takes a long time. In fig2 of the paper Portrait4Dv2, I cannot find where the FLAME is used in. And from the code, I find it is used to construct the Deformer, but I do not know the deformer is built for what. Moreover, if we only want to get certain view(e.g. just the front view) from reference image driven by target image, do i still need the FLAME params? If not, the process would be much faster. Thanks!

The Deformer constructed by FLAME is only used for neck pose control. If you only want frontal view, or rendering novel camera views without neck pose variation, it is just fine to set all FLAME parameters to zeros.

**Comment by longyangqi** [comment]
> The Deformer constructed by FLAME is only used for neck pose control. If you only want frontal view, or rendering novel camera views without neck pose variation, it is just fine to set all FLAME parameters to zeros.

Thanks for you rapid reply！

**Comment by Pixie8888** [comment]
Dear @YuDeng , 

I also have a question about neck pose params in flame model. I am trying to do visualization of fitted flame model in ```data_preprocess/flame_optim_batch_singleframe.py```. Why does the flame model always take in zero ```neck_pose_params``` as shown below:
![image](https://github.com/user-attachments/assets/c8d732e1-0618-452e-a3c7-c3507484a130)

If I add the ```neck_pose_params``` (removing ```torch.zeros_like```), it generates wrong 3D head as shown below:
![image](https://github.com/user-attachments/assets/334b1e5a-3b7a-4d04-a8a7-e18217577b42)


If I follow the zero neck_pose_params as written in the code, it gives reasonable 3D head as shown below:
![image](https://github.com/user-attachments/assets/337c8ec6-6a16-463e-a91b-1e0254044c55)

Why do we need to use zero neck_pose_params?

**Comment by Pixie8888** [comment]
ok,I find the reason. It is because the ```c_compose``` contains neck pose.
